Root.tsx

```diff
@@ -0,0 +1,86 @@
+import 'react-native-gesture-handler';
+import React, { useContext, useEffect, useState } from 'react';
+import { ActivityIndicator, View, StyleSheet } from 'react-native';
+import { NavigationContainer } from '@react-navigation/native';
+import { createStackNavigator } from '@react-navigation/stack';
+import HomeScreen from './screens/HomeScreen';
+import { AuthContext, AuthContextType } from './context/AuthProvider';
+import LoginScreen from './screens/LoginScreen';
+import * as SecureStore from 'expo-secure-store';
+
+const Stack = createStackNavigator();
+
+const HomeStackNavigator = () => {
+  return (
+    <Stack.Navigator>
+      <Stack.Screen
+        name="Home Screen"
+        component={HomeScreen}
+        options={{ title: 'Home' }}
+      />
+    </Stack.Navigator>
+  );
+};
+
+const AuthStackNavigator = () => {
+  return (
+    <Stack.Navigator
+      screenOptions={{ headerShown: false, headerBackTitleVisible: false }}
+    >
+      <Stack.Screen
+        name="Login Screen"
+        component={LoginScreen}
+        options={{ headerShown: false }}
+      />
+    </Stack.Navigator>
+  );
+};
+
+export default function App() {
+  const [isLoading, setIsLoading] = useState<boolean>(true);
+  const { user, setUser } = useContext<AuthContextType>(AuthContext);
+
+  useEffect(() => {
+    SecureStore.getItemAsync('user')
+      .then(userString => {
+        if (userString) {
+          setUser(JSON.parse(userString));
+        }
+        setIsLoading(false);
```

Comment:
you could do just one `setIsLoading(false)` in a `finally()` instead of having to do it in the `then()` AND in the `catch()`